vnext/Microsoft.ReactNative/Views/ShadowNodeBase.cpp

```diff
@@ -162,4 +162,8 @@ void ShadowNodeBase::YellowBox(const std::string &message) const noexcept {
   GetViewManager()->GetReactContext().CallJSFunction("RCTLog", "logToConsole", folly::dynamic::array("warn", message));
 }
 
+void ShadowNodeBase::RedBox(const std::string &message) const noexcept {
```

Comment:
I know you didn't add this, but having these methods on ShadowNodeBase isn't very cohesive. It creates a convenience for certain functions in certain parts of ViewManagers that cannot be used generically, and isn't part of the ViewManager interface.

A pattern to allow these to be used more generically could be free functions that accept a `ReactContext`. That would allow ViewManagers or other bits of the code to trigger these errors in a generic way, Depending on your comfort level, feel free to leave this as is and file an issue for someone to take a look.

Reply:
I see that makes sense. And would that free function continue to live in ShadowNodeBase? I guess the one other question I have in my understanding is that CallJSFunction expects to be called by an IReactContext but NativeUIManager for example uses a ReactContext. Is there a way to convert from a ReactContext to a IReactContext? Or is there an equivalent for CallJSFunction for ReactContexts?

Reply:
It would probably make sense to live in a separate file somewhere. It's probably fine to leave this as is for now though.

Re the difference between ReactContext and IReactContext, the former is a wrapper around the latter. The IReactContext interface is generated from an IDL file and has certain restrictions, where the wrapper is the more user friendly API.

You can get the underlying IReactContext from the ReactContext by calling its "Handle" function, and you can go the other way by passing an IReactContext to the ReactContext constructor. This pattern is present in a few other places in this code.